docs/addons/ssm-agent.md

```diff
@@ -0,0 +1,97 @@
+# SSM Agent Add-on
+
+This add-on uses the Kubernetes DaemonSet resource type to install AWS Systems Manager Agent (SSM Agent) on all worker nodes, instead of installing it manually or replacing the Amazon Machine Image (AMI) for the nodes. DaemonSet uses a CronJob on the worker node to schedule the installation of SSM Agent.
+
+A common use-case for installing SSM Agent on the worker nodes is to be able open a terminal session on an instance without the need to create a bastion instance and without having to install SSH keys on the worker nodes.
+
+The AWS Identity and Access Management (IAM) managed role **AmazonSSMManagedInstanceCore** provides the required permissions for SSM Agent to run on EC2 instances. This role is automatically attached to the instances when this add-on is enabled.
+
+**Limitations**
+
+* This add-on isn't applicable to AWS Fargate, because DaemonSets aren't supported on the Fargate platform.
+
+* This add-on applies only to Linux-based worker nodes.
+
+* The DaemonSet pods run in privileged mode. If the Amazon EKS cluster has a webhook that blocks pods in privileged mode, the SSM Agent will not be installed.
+
+* Only latest version of SSM Agent add-on can be installed.
+
+## Usage
+
+```typescript
+import { SSMAgentAddOn }  from '@shapirov/cdk-eks-blueprint';
+
+readonly ssmAgent = new SSMAgentAddon();
+
+const addOns: Array<ClusterAddOn> = [ ssmAgent ];
+
+const app = new cdk.App();
+new EksBlueprint(app, 'my-stack-name', addOns, [], {
+  env: {
+      account: <AWS_ACCOUNT_ID>,
+      region: <AWS_REGION>,
+  },
+});
+```
+
+To validate that SSM Agent is running on worker node instance:
+
+1. Get the EC2 Instance Id of a worker node
+
+```bash
+instance_id=$(kubectl get nodes -o custom-columns=NAME:.metadata.name,INSTANCEID:.spec.providerID | awk -F/ 'FNR == 2 {print $5}')
+```
+
+2. Use the `start-session` api to see if you can open a terminal into the instance
+
+```bash
+aws ssm start-session --target $instance_id
```

Comment:
@askulkarni2 during validation, I ran into an issue where I didn't have the `Session Manager plugin for the AWS CLI installed`. We should add that as a step here. 

https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with-install-plugin.html

Reply:
addressed